### PR TITLE
Ensure that trainees can edit their profile details

### LINKED ISSUE #1095: Permissions system: "admin"-considered groups don't grant template-scope permissions
In [`models.py:545`](https://github.com/swcarpentry/amy/blob/develop/workshops/models.py#L545) we have a `is_admin` function that determines if person should have admin privileges based on groups they're in. For example, any trainer is considered admin.

In the templates, however, we explicitly check for permissions, like `workshops.change_person` in [`workshops/templates/workshops/person.html`](https://github.com/swcarpentry/amy/blob/develop/workshops/templates/workshops/person.html#L10). The issue is, this check fails for people in `trainers` group because, even though `trainers` is considered an admin group, it doesn't have the `workshops.change_person` permission.

### PR DESCRIPTION
**Pull request by narayanaditya95**
Changes:
  - Rewrite `badge_award` view to `award_add` view to handle
    addition of awards. Supports prepopulation of the `badge`
    and `person` fields.
  - Refactor the `person_edit` view into a CBV.
    The forms within the *Awards* and *Tasks* tab submit to their
    respective Create views.
  - Trainees page now redirects to `award_add` view to allow
    awarding DC and SWC badges to trainees.
    Maintains all previous functionality.

@pbanaszkiewicz @chrismedrela Please review.
I am rewriting certain failing tests with `django-webtest` in the meanwhile.

Fixes #1095

### CODE REVIEW

**Comment by pbanaszkiewicz** [comment]
@narayanaditya95 how's your test rewriting going?

File: workshops/views.py [hunk 1]
```diff
@@ -683,126 +686,29 @@ def form_valid(self, form):
...
+    model = Person
+    form_class = PersonForm
+    pk_url_kwarg = 'person_id'
+    template_name = 'workshops/person_edit_form.html'
 
-        else:
-            messages.error(request, 'Fix errors below.')
+    def test_func(self):
+        return self.request.user.has_perm('workshops.edit_person') or \
+            self.request.user == self.get_object()
```
Comment (pbanaszkiewicz) [inline workshops/views.py]: It's for the additional `UserPassesTestMixin` mixin.

File: workshops/models.py [hunk 2]
```diff
@@ -1653,6 +1653,9 @@ class Meta:
     def __str__(self):
         return '{0}/{1}/{2}/{3}'.format(self.person, self.badge, self.awarded, self.event)
 
+    def get_absolute_url(self):
+        return reverse('badge_details', args=[self.badge.name])
```
Comment (narayanaditya95) [inline workshops/models.py]: It is rather the list view.
Sorry about that, reverting.

File: workshops/views.py (hunk 1 above)
Comment (narayanaditya95) [inline workshops/views.py]: ^Yes

File: workshops/views.py [hunk 3]
```diff
@@ -683,126 +686,29 @@ def form_valid(self, form):
...
-                ),
-            )
-
-            return redirect(person)
+class PersonUpdate(OnlyForAdminsMixin, UserPassesTestMixin,
+                   AMYUpdateView):
+    model = Person
+    form_class = PersonForm
+    pk_url_kwarg = 'person_id'
+    template_name = 'workshops/person_edit_form.html'
```
Comment (narayanaditya95) [inline workshops/views.py]: So, the current permission logic is:
- A user can access the `person_edit` view if he is an admin user and has the `workshops.edit_person` permission or he/she is the person being edited.
- If a person satisfies the above criteria, he can access the view. There is another layer of permission handling within the templates.
    - If the person has the `workshops.edit_award` or `workshops.edit_task` permissions, the respective award and task add forms are rendered. Else, he can only view the list of awards and tasks.

The explicit permissions are handled within the templates. Am I missing something?

**Comment by narayanaditya95** [comment]
@pbanaszkiewicz Up for review.

**Comment by pbanaszkiewicz** [comment]
@narayanaditya95 thanks for the changes and the explanation. Can you please split this big commit into smaller chunks?

**Comment by narayanaditya95** [comment]
I can divide it into two commits, one representing the refactor w/o addressing issue #1095 and other the fixes it. If I divide it further, those chunks would not be working instances of AMY.

**Comment by narayanaditya95** [comment]
@pbanaszkiewicz Up for final review. :)

**Comment by chrismedrela** [comment]
@narayanaditya95 I really like your PR, especially that the codebase is smaller now. :-) One question: why do we need two classes: `MockAwardCreate` and `AwardCreate` instead of one?

**Comment by chrismedrela** [comment]
@narayanaditya95, I got it, please ignore me.

**Pull request merged**